casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -154,23 +154,15 @@ object Validate {
     }
   }
 
-  /*
-   * TODO: Double check ordering of validity checks
-   */
   def blockSummary[F[_]: Sync: Log: Time: BlockStore: BlockDagStorage: Metrics: Span](
       block: BlockMessage,
-      genesis: BlockMessage,
       s: CasperSnapshot,
       shardId: String,
       expirationThreshold: Int
   ): F[ValidBlockProcessing] =
     (for {
-      _ <- EitherT.liftF(Span[F].mark("before-block-hash-validation"))
-      _ <- EitherT(Validate.blockHash(block))
       _ <- EitherT.liftF(Span[F].mark("before-timestamp-validation"))
       _ <- EitherT(Validate.timestamp(block))
-      _ <- EitherT.liftF(Span[F].mark("before-shard-identifier-validation"))
-      _ <- EitherT(Validate.shardIdentifier(block, shardId))
```

Comment:
Why `shardId` validation for block was removed?

Reply:
https://github.com/rchain/rchain/pull/3672/commits/fd688400030326ac3e1f8173c735455d20a3fe63#r860065262 its should not slashable offence but just a reason to drop block on the floor I guess